### PR TITLE
Add support for Jakarta CDI

### LINKED ISSUE #2950: Support for Jakarta @ApplicationScoped is missing
https://jakarta.ee/specifications/cdi/3.0/jakarta-cdi-spec-3.0.html#bean_defining_annotations defines a bunch of bean defining annotations. 

Prior to Jakarta, 'cdi' component model on a mapper resulted in an @ApplicationScoped implementation class, which could get properly resolved.

However, with Jakarta CDI spec 3.0, @ApplicationScoped no longer 'sits' in javax. but in jakarta. The result of this is that mapstruct cannot be used in such an arrangement.

### PR DESCRIPTION
**Pull request by filiphr**
Fixes #2950

### CODE REVIEW

File: processor/src/main/java/org/mapstruct/ap/internal/processor/CdiComponentProcessor.java [hunk 1]
```diff
@@ -48,4 +50,24 @@ protected boolean requiresGenerationOfDecoratorClass() {
     protected boolean additionalPublicEmptyConstructor() {
         return true;
     }
+
+    private Type getType(String simpleName) {
```
Comment (sjaakd) [inline processor/src/main/java/org/mapstruct/ap/internal/processor/CdiComponentProcessor.java:54]: `getCDIType`

**Review by sjaakd — APPROVED** [review]

File: processor/src/main/java/org/mapstruct/ap/internal/processor/CdiComponentProcessor.java (hunk 1 above)
Comment (filiphr) [reply processor/src/main/java/org/mapstruct/ap/internal/processor/CdiComponentProcessor.java:54]: I didn't want to complicate things. This also fetches the `@Inject` from the inject package as well

**Comment by Felk** [comment]
Hello, I am curious if this is landing in a release any time soon? I am playing around with [Quarkus 3.0.0 Alpha4](https://github.com/quarkusio/quarkus/releases/tag/3.0.0.Alpha4), which includes the transition from the `javax` to `jakarta` namespace. Being able to just switch the component model from `CDI` to `JAKARTA_CDI` would mean mapstruct is well-prepared with a clear migration path once Quarkus 3.0 hits stable 😁

**Comment by edeandrea** [comment]
I too am looking for this feature because of Quarkus 3 & the move to the jakarta namespace. RIght now I really have no workaround as `@Singleton` does not work (because I mock/spy using Mockito) and you can't mock/spy on `@Singleton` beans. It needs to be `@ApplicationScoped`.

**Comment by edeandrea** [comment]
@filiphr / @sjaakd Please see previous comments.

**Comment by filiphr** [comment]
This has now been backported to 1.5.4

**Pull request merged**